src/client.h

```diff
@@ -175,6 +176,10 @@ struct ClientEvent
 			std::string *formspec;
 			std::string *formname;
 		} show_formspec;
+		struct{
+			const char* formspec;
```

Comment:
you can't do that, `std::string` inside a union causes errors, at the very least my compiler can't handle it.